### DISCUSSION TITLE
Relative abundance from comp_barplot showed 10 instead of 1

### DISCUSSION THREAD

**Discussion by limkahkheng** [david-barnett/microViz#179] [General]
Hi all,

I'm a beginner of this coding.

I tried to run the code as outlined in the tutorial in this link:[ ](https://david-barnett.github.io/microViz/articles/web-only/compositions.html). However, I don't know why the y-axis of the Relative abundance is 10.0 instead of 1.0. Anyone experienced the same problem?

Here is my code, mifish_chord is a phyloseq object:
```r
mifish_wat<-mifish_chord %>%
  ps_filter(Medium == "Water") %>% # only one sample in this group
  ps_mutate(
    Volume = factor(Volume, levels = c("1","2","3","4")),
    Site = factor(Site, levels = unique(Site))
  ) %>%
  comp_barplot(
    tax_level = "genus", n_taxa = 20, sample_order="asis", bar_width = 0.7, merge_other = FALSE, x = "Volume"
  ) +
  facet_wrap(facets = vars(Site)) +
  theme_bw() + # slightly clearer axes for facets
  labs(x = "Volume", y = "Relative Abundance", fill = "genus") +
  geom_bar(position = "stack", stat = "identity") +
  scale_y_continuous(
    expand = expansion(add = c(0, 0.1))
  )
mifish_wat
```
![image](https://github.com/user-attachments/assets/e654eaf2-cc6b-4969-8518-236db9c14c28)

**Comment by david-barnett** [comment]
Hi, when you set a custom x axis variable, you need to ensure (using your faceting strategy) that there will only be one sample per bar.

So I guess you have 10 samples within each Volume and Site combination, so you are expecting 80 samples? You'll need to add another variable to the facet_wrap, or try facet_grid maybe.

Howver, if you do only expect 8 samples, then it would be a different problem.

**Reply by limkahkheng** [reply]
Thanks David for your prompt reply.

You're right, I do have 10 samples for each Volume and Site combination. I've tried the alternatives that you suggested yet the relative abundance is still 10. We decided to present to sample individually (in fact they are just the 9 other technical replicates) and the relative abundance became 1. At least we know it's x-axis variable that causing the abnormally.
